### PR TITLE
feat(wallet): use LiFi provider for EVM swaps

### PR DESCRIPTION
**Pull request by onyb**
Resolves https://github.com/brave/brave-browser/issues/36436

⚠️ Merge after https://github.com/brave/brave-core/pull/23730, or at least ensure they land in the same version.

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

Test EVM swaps for all supported networks, including new ones such as Gnosis, Polygon zkEVM, etc.

### CODE REVIEW

**Review by Douglashdaniel — APPROVED** [review]
++ Wallet Front-end

File: components/brave_wallet/browser/swap_service.cc [hunk 1]
```diff
@@ -383,42 +383,53 @@ void SwapService::GetQuote(mojom::SwapQuoteParamsPtr params,
...
+                                GetZeroExQuoteURL(*params, fee_param), "", "",
                                 std::move(internal_callback), GetHeaders(), {},
                                 std::move(conversion_callback));
 
     return;
   }
 
-  if (IsNetworkSupportedByLiFi(params->from_chain_id) &&
-      IsNetworkSupportedByLiFi(params->to_chain_id)) {
+  // EVM swaps are served via LiFi if the provider is set to kLiFi or kAuto.
```
Comment (supermassive) [inline components/brave_wallet/browser/swap_service.cc:429]: Could you please reorder these back to `ZeroEx, Jupiter, LiFi`? Would be a bit easier to review

Comment (onyb) [reply components/brave_wallet/browser/swap_service.cc]: Done

File: ios/brave-ios/Sources/BraveWallet/Crypto/Stores/SwapTokenStore.swift [hunk 2]
```diff
@@ -367,7 +367,8 @@ public class SwapTokenStore: ObservableObject, WalletObserverStore {
       toToken: buyAddress,
       toAmount: buyAmountInWei,
       slippagePercentage: "\(slippagePercentage)",
-      routePriority: .recommended
+      routePriority: .recommended,
+      provider: .auto
```
Comment (StephenHeaps) [inline ios/brave-ios/Sources/BraveWallet/Crypto/Stores/SwapTokenStore.swift:371]: @onyb maybe better we check token coin and force `.zeroX` for Ethereum / `.jupiter` for Solana until we have [updated copy](https://github.com/brave/brave-core/blob/9dd6442173b64d26804ebe8c26ef0f4d177a1bec/ios/brave-ios/Sources/BraveWallet/Crypto/WalletActions/SwapCryptoView.swift#L247) / decision for disclaimers. Unless this PR is waiting for that decision?

```suggestion
      // TODO(stephenheaps): Enable Lifi support https://github.com/brave/brave-browser/issues/36436
      provider: sellToken.coin == .sol ? .jupiter : .zeroEx
```

**Review by supermassive — APPROVED** [review]
wallet core lgtm

**Review by StephenHeaps — APPROVED** [review]
iOS++

**Pull request merged**